joomla/joomla-cms
# [4.0] Flex based control-group fields

Pull Request for Issue #27558, #27681, #27676 #27664 .

### Summary of Changes
Moves field control-groups to flex.

- Less CSS required
- Flex naturally supports RTL so no RTL CSS required.
- Media queries look at the width of the viewport and are the only way to change the orientation of the current control-group fields. Flex would allow us to change the orientation of the field depending on the space available to it. A much more versatile and safer way to approach responsive fields especially.
- Duplicate CSS removed
- Fixes `form-vertical` class

Simply put, with this PR,  label/fields will become container aware,  moving from vertical to horizontal aligned if the field is reduced to less than 210px. I have also enabled the `form-vertical` class. The addition of this class and the field will always display vertically. The `fields-no-margin` has been removed completely.

Edit: Any field in a multi-column layout, displays in a vertical format regardless of the space available to it. This was always the case.

### Testing Instructions
Apply this patch and run `node build.js --compile-css` for updating the changed SCSS. Check fields display correctly.



### Documentation Changes Required
